src/test/scala/org/broadinstitute/dsde/firecloud/dataaccess/MockRawlsDAO.scala

```diff
@@ -70,6 +70,32 @@ class MockRawlsDAO  extends RawlsDAO {
     false
   )
 
+  private val unpublishedRawlsWorkspaceLibraryValid = Workspace(
+    "attributes",
+    "att",
+    None, //realm
+    "id",
+    "", //bucketname
+    DateTime.now(),
+    DateTime.now(),
+    "ansingh",
+    Map(AttributeName("default", "a") -> AttributeBoolean(true),
+      AttributeName("default", "b") -> AttributeNumber(1.23),
+      AttributeName("default", "c") -> AttributeString(""),
+      AttributeName("library", "invalidDataset") -> AttributeBoolean(false),
+      AttributeName("library", "projectName") -> AttributeString("testing"),
+      AttributeName("default", "d") -> AttributeString("escape quo\"te"),
+      AttributeName("default", "e") -> AttributeString("v1"),
+      AttributeName("default", "f") -> AttributeValueList(Seq(
+        AttributeString("v6"),
+        AttributeNumber(999),
+        AttributeBoolean(true)
+      ))),
+    Map(), //acls
+    Map(), //realm acls,
+    false
+  )
```

Comment:
optional feedback: since `Workspace` is a case class, you can use `copy()` ... something like:
```
  private val unpublishedRawlsWorkspaceLibraryValid = publishedRawlsWorkspaceWithAttributes.copy(
    attributes = publishedRawlsWorkspaceWithAttributes.attributes +
      (AttributeName.withLibraryNS("invalidDataset") -> AttributeBoolean(false)) - 
      AttributeName.withLibraryNS("published"))
```

also, I know you're just copying/pasting this code, but I prefer `AttributeName.withDefaultNS` / `AttributeName.withLibraryNS` instead of hardcoding "library" and "default".